### PR TITLE
JDK-8344057 : Remove doPrivileged calls from unix platform sources in the java.desktop module

### PR DESCRIPTION
**Pull request by honkar-jdk**
Post JEP-486 (Permanently Disable the Security Manager) cleanup.
Calls to java.security.AccessController.doPrivileged are obsolete thus removed in this PR.

This PR addresses removal of AccessController.doPrivileged() calls from unix-platform files in the java.desktop module. Any SM related imports that are no longer needed are removed.

This PR is limited to removing doPrivileged() calls and excludes any refactoring, reformatting, or other clean up that is out-of-scope for this fix.

PS: I have explicitly add comments to the changes where a more watchful review is required.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344057](https://bugs.openjdk.org/browse/JDK-8344057): Remove doPrivileged calls from unix platform sources in the java.desktop module (**Bug** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22218/head:pull/22218` \
`$ git checkout pull/22218`

Update a local copy of the PR: \
`$ git checkout pull/22218` \
`$ git pull https://git.openjdk.org/jdk.git pull/22218/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22218`

View PR using the GUI difftool: \
`$ git pr show -t 22218`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22218.diff">https://git.openjdk.org/jdk/pull/22218.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22218#issuecomment-2484177943)
</details>

### CODE REVIEW

File: src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java [hunk 1]
```diff
@@ -370,28 +360,24 @@ public void dispatchEvent(XEvent ev) {
...
+            if (xs != null) {
+                ((XAWTXSettings)xs).dispose();
+            }
+            freeXKB();
+            if (log.isLoggable(PlatformLogger.Level.FINE)) {
+                dumpPeers();
+            }
         };
-        AccessController.doPrivileged(a);
+        String name = "XToolkt-Shutdown-Thread";
```
Comment (prrace) [inline src/java.desktop/unix/classes/sun/awt/X11/XToolkit.java:376]: I am noting that there's a pre-existing typo here. Not asking you to fix it.

File: src/java.desktop/unix/classes/sun/print/UnixPrintJob.java [hunk 2]
```diff
@@ -525,8 +524,6 @@ public void print(Doc doc, PrintRequestAttributeSet attributes)
 
         // now spool the print data.
         PrinterOpener po = new PrinterOpener();
-        @SuppressWarnings("removal")
-        var dummy = java.security.AccessController.doPrivileged(po);
```
Comment (prrace) [inline src/java.desktop/unix/classes/sun/print/UnixPrintJob.java:529]: You are no longer running the action

**Review by prrace — CHANGES_REQUESTED** [review]

File: src/java.desktop/unix/classes/sun/print/UnixPrintJob.java (hunk 2 above)
Comment (honkar-jdk) [reply src/java.desktop/unix/classes/sun/print/UnixPrintJob.java:529]: I'm not sure if I understood it correctly: Did you mean remove `implements java.security.PrivilegedAction<Object>` from PrinterOpener class ?

File: src/java.desktop/unix/classes/sun/awt/X11GraphicsEnvironment.java [hunk 3]
```diff
@@ -59,78 +59,71 @@ public final class X11GraphicsEnvironment extends SunGraphicsEnvironment {
...
-                    // only attempt to initialize GLX if it was requested
-                    if (glxRequested) {
-                        glxAvailable = initGLX();
-                        if (glxVerbose && !glxAvailable) {
-                            System.out.println(
-                                "Could not enable OpenGL " +
-                                "pipeline (GLX 1.3 not available)");
-                        }
-                    }
+                if(xProp.equalsIgnoreCase("t") || xProp.equalsIgnoreCase("true")) {
```
Comment (turbanoff) [inline src/java.desktop/unix/classes/sun/awt/X11GraphicsEnvironment.java:96]:
```suggestion
                if (xProp.equalsIgnoreCase("t") || xProp.equalsIgnoreCase("true")) {
```

File: src/java.desktop/unix/classes/sun/awt/X11/XFileDialogPeer.java [hunk 4]
```diff
@@ -151,12 +149,7 @@ private void init(FileDialog target) {
         savedDir = target.getDirectory();
         // Shouldn't save 'user.dir' to 'savedDir'
         // since getDirectory() will be incorrect after handleCancel
-        userDir = AccessController.doPrivileged(
-            new PrivilegedAction<String>() {
-                public String run() {
-                    return System.getProperty("user.dir");
-                }
-            });
+        userDir =  System.getProperty("user.dir");
```
Comment (turbanoff) [inline src/java.desktop/unix/classes/sun/awt/X11/XFileDialogPeer.java:152]:
```suggestion
        userDir = System.getProperty("user.dir");
```

File: src/java.desktop/unix/classes/sun/print/UnixPrintJob.java (hunk 2 above)
Comment (prrace) [reply src/java.desktop/unix/classes/sun/print/UnixPrintJob.java:529]: Well, that too. But the main point is WHERE is the code now that calls PrinterOpener.run() ?
The thing that does the actual work. So far as I can see you've deleted the call and done nothing to replace it.

Comment (prrace) [reply src/java.desktop/unix/classes/sun/print/UnixPrintJob.java:529]: So 2 things to do that I can see.
1) after line 526 call po.run();
2) Go to the definition of PrinterOpener and remove implements java.security.PrivilegedAction<Object>

Anything more would be too much refactoring.

Same idea below for PrinterSpooler.

How were you testing this ?

Comment (honkar-jdk) [reply src/java.desktop/unix/classes/sun/print/UnixPrintJob.java:529]: > after line 526 call po.run();
> Go to the definition of PrinterOpener and remove implements java.security.PrivilegedAction
> Anything more would be too much refactoring.

Makes sense. Thank you for clarifying.

> How were you testing this ?

Ran CI tests - clientlibs, wayland jobs for these unix file changes as of now. I believe to test this properly we need to run manual printer tests.

**Review by prrace — APPROVED** [review]

File: src/java.desktop/unix/classes/sun/awt/screencast/TokenStorage.java [hunk 5]
```diff
@@ -317,7 +303,7 @@ private static void storeTokenFromNative(String oldToken,
             }
 
             if (changed) {
-                doPrivilegedRunnable(() -> store(PROPS_PATH, "save tokens"));
+               store(PROPS_PATH, "save tokens");
```
Comment (turbanoff) [inline src/java.desktop/unix/classes/sun/awt/screencast/TokenStorage.java:306]:
```suggestion
                store(PROPS_PATH, "save tokens");
```

File: src/java.desktop/unix/classes/sun/print/PrintServiceLookupProvider.java [hunk 6]
```diff
@@ -886,51 +871,47 @@ static String[] execCmd(final String command) {
...
+                }
+                proc.getInputStream()
+                    .close();
+                proc.getErrorStream()
+                    .close();
+                proc.getOutputStream()
+                    .close();
+            }
+        } catch (IOException io) {
+            io.printStackTrace();
```
[the remark this answers is not in the document]
Comment (honkar-jdk) [reply src/java.desktop/unix/classes/sun/print/PrintServiceLookupProvider.java:914]: @prrace Can you please review this code snippet? Is it okay to catch the IOException here or propagate it ?

[the remark this answers is not in the document]
Comment (prrace) [reply src/java.desktop/unix/classes/sun/print/PrintServiceLookupProvider.java:914]: The existing code caught it and did not propagate it.
So it should be caught and not propagated - and do not add the printStackTrace.
remove it.

[the remark this answers is not in the document]
Comment (honkar-jdk) [reply src/java.desktop/unix/classes/sun/print/PrintServiceLookupProvider.java:914]: Updated. I didn't notice it earlier but the original exception `PrivilegedExceptionAction` was being caught but not re-thrown which now can be replace with IOException in similar way.

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**